examples/jsm/exporters/GLTFExporter.js

```diff
@@ -2540,6 +2540,10 @@ class GLTFWriter {
 
 		if ( objectsWithoutScene.length > 0 ) {
 
+			if ( input.length === 1 ) {
+				this.extensionsUsed[ 'GODOT_single_root' ] = true;
```

Comment:
I'm not sure about automatically use `GODOT_single_root` with just one 3D object. That might break the exporter for developers who rely on the current behavior. I would suggest 

- only update `GLTFLoader` for now.
- or enable usage of `GODOT_single_root` with a new exporter option
- or make use of `userData.gltfExtensions` which would be defined for the single 3D object. If that works with `GODOT_single_root`, that is probably sufficient for now.
https://github.com/mrdoob/three.js/blob/5bf53eec7836b1dc4ebaff478929e1970c258d92/examples/jsm/exporters/GLTFExporter.js#L3569

Reply:
I don't think `userData.gltfExtensions` is appropriate, because that would place `"extensions": { "GODOT_single_root: {} }` on a glTF node, which is incorrect.

I would argue it should be enabled by default, since it significantly improves behavior when the flag is recognized, has no effect when unrecognized, and is a flag users likely won't know to reach out for. But sure I can change this. I've pushed a commit that changes this to an exporter option, false by default.

Reply:
Actually, I updated the PR again, this time completely removing the export code.

Three.js exports the root node as the last node, while GODOT_single_root requires that the node at index 0 is the root node. Before we can flag GODOT_single_root, we would need to change the export code to place the nodes in a different order. That is worth changing as its own PR separately from GODOT_single_root. The current status quo is that UnityGLTF and Godot both put the root at index 0, but Three.js and Blender put it as the last node.